Some kind of go error/panic when doing cluster add

```
$ flynn cluster add -p <Base64 string> default cluster.bemt.eu <key/hash> -f
panic: runtime error: slice bounds out of range [recovered]
	panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/flynn/flynn/pkg/shutdown.(*handler).exit(0xc2080d0810, 0x0, 0x0, 0x0, 0x7e30e0, 0xc208032010)
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/pkg/shutdown/handler.go:69 +0x12e
github.com/flynn/flynn/pkg/shutdown.Exit()
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/pkg/shutdown/handler.go:40 +0x73
github.com/flynn/flynn/cli/config.(*Config).Add(0xc2080d1d10, 0xc208055300, 0x8d5301, 0x0, 0x0)
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/cli/config/config.go:118 +0x630
main.runClusterAdd(0xc2080ec580, 0x0, 0x0)
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/cli/cluster.go:95 +0x5d4
main.runCluster(0xc2080ec580, 0x0, 0x0)
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/cli/cluster.go:51 +0x109
main.runCommand(0x7fff5fbffbce, 0x7, 0xc208054e80, 0x7, 0x8, 0x0, 0x0)
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/cli/main.go:176 +0x8ba
main.main()
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/cli/main.go:122 +0xd1a

goroutine 5 [syscall]:
os/signal.loop()
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/util/_toolchain/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/util/_toolchain/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 11 [runnable]:
github.com/flynn/flynn/pkg/shutdown.(*handler).wait(0xc2080d0810)
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/pkg/shutdown/handler.go:55
created by github.com/flynn/flynn/pkg/shutdown.newHandler
	/tmp/tmp.NETonRerUm/src/github.com/flynn/flynn/pkg/shutdown/handler.go:25 +0xa8
```
```
$ flynn --version
v20151001.0
```
I have a "default" cluster already but I want to overwrite that config. If I change "default" to another name (that also exists in ~/.flynnrc) the command executes as properly.

--- a/cli/config/config.go
+++ b/cli/config/config.go
@@ -94,9 +94,9 @@ func (c *Config) Marshal() []byte {
 }
 
 func (c *Config) Add(s *Cluster, force bool) error {
+	var msg string
+	conflictIdx := -1
 	for i, existing := range c.Clusters {
-		msg := ""
-
 		switch {
 		case existing.Name == s.Name:
 			msg = fmt.Sprintf("Cluster %q already exists in ~/.flynnrc", s.Name)
@@ -107,19 +107,24 @@ func (c *Config) Add(s *Cluster, force bool) error {
 		case existing.GitHost != "" && existing.GitHost == s.GitHost:
 			msg = fmt.Sprintf("A cluster with the git host %q already exists in ~/.flynnrc", s.GitHost)
 		}
-
-		// The new cluster config match with existing one
 		if msg != "" {
-			if !force {
-				return fmt.Errorf(msg)
+			if conflictIdx != -1 && conflictIdx != i {
+				return fmt.Errorf("The cluster name and/or URLs conflict with multiple existing clusters.")
 			}
-
-			// Remove existing match
-			c.Clusters = append(c.Clusters[:i], c.Clusters[i+1:]...)
+			conflictIdx = i
 		}
 	}
 
-	// Did not match
+	// The new cluster config conflicts with an existing one
+	if msg != "" {
+		if !force {
+			return fmt.Errorf(msg)
+		}
+
+		// Remove conflicting cluster
+		c.Clusters = append(c.Clusters[:conflictIdx], c.Clusters[conflictIdx+1:]...)
+	}
+
 	c.Clusters = append(c.Clusters, s)
 
 	return nil

--- a/test/test_cli.go
+++ b/test/test_cli.go
@@ -615,6 +615,8 @@ func (s *CLISuite) TestCluster(t *c.C) {
 	}
 
 	// cluster add
+	t.Assert(flynn("cluster", "add", "-g", "foo.example.com:2222", "foo", "https://controller.foo.example.com", "e09dc5301d72be755a3d666f617c4600"), Succeeds)
+	t.Assert(flynn("cluster"), SuccessfulOutputContains, "foo")
 	t.Assert(flynn("cluster", "add", "-g", "test.example.com:2222", "-p", "KGCENkp53YF5OvOKkZIry71+czFRkSw2ZdMszZ/0ljs=", "test", "https://controller.test.example.com", "e09dc5301d72be755a3d666f617c4600"), Succeeds)
 	t.Assert(flynn("cluster"), SuccessfulOutputContains, "test")
 	t.Assert(flynn("cluster", "add", "-f", "-g", "test.example.com:2222", "-p", "KGCENkp53YF5OvOKkZIry71+czFRkSw2ZdMszZ/0ljs=", "test", "https://controller.test.example.com", "e09dc5301d72be755a3d666f617c4600"), Succeeds)
@@ -625,8 +627,13 @@ func (s *CLISuite) TestCluster(t *c.C) {
 	cfg, err := config.ReadFile(file.Name())
 	t.Assert(err, c.IsNil)
 	t.Assert(cfg.Default, c.Equals, "test")
-	t.Assert(cfg.Clusters, c.HasLen, 1)
-	t.Assert(cfg.Clusters[0].Name, c.Equals, "test")
+	t.Assert(cfg.Clusters, c.HasLen, 2)
+	t.Assert(cfg.Clusters[0].Name, c.Equals, "foo")
+	t.Assert(cfg.Clusters[1].Name, c.Equals, "test")
+	// overwriting with a conflicting name and a different conflicting url should error
+	conflict := flynn("cluster", "add", "-f", "-g", "test.example.com:2222", "foo", "https://controller.test.example.com", "e09dc5301d72be755a3d666f617c4600")
+	t.Assert(conflict, c.Not(Succeeds))
+	t.Assert(conflict, OutputContains, "conflict with")
 	// overwriting (without --force) should not work
 	t.Assert(flynn("cluster", "add", "test", "foo", "bar"), c.Not(Succeeds))
 	t.Assert(flynn("cluster"), SuccessfulOutputContains, "test")
@@ -643,7 +650,8 @@ func (s *CLISuite) TestCluster(t *c.C) {
 	t.Assert(flynn("cluster"), c.Not(SuccessfulOutputContains), "test")
 	cfg, err = config.ReadFile(file.Name())
 	t.Assert(err, c.IsNil)
-	t.Assert(cfg.Clusters, c.HasLen, 0)
+	t.Assert(cfg.Clusters, c.HasLen, 1)
+	t.Assert(flynn("cluster", "remove", "foo"), Succeeds)
 	// cluster remove default and set next available
 	t.Assert(flynn("cluster", "add", "-d", "-g", "test.example.com:2222", "-p", "KGCENkp53YF5OvOKkZIry71+czFRkSw2ZdMszZ/0ljs=", "test", "https://controller.test.example.com", "e09dc5301d72be755a3d666f617c4600"), Succeeds)
 	t.Assert(flynn("cluster", "add", "-g", "next.example.com:2222", "-p", "KGCENkp53YF5OvOKkZIry71+czFRkSw2ZdMszZ/0ljs=", "next", "https://controller.next.example.com", "e09dc5301d72be755a3d666f617c4600"), Succeeds)

